### ISSUE TITLE
Manage releases of starters

### ISSUE THREAD

**Issue by JohnAlbin** [chapter-three/next-drupal#628]
### Package

basic-starter, graphql-starter

### Describe the feature request

The `next-drupal` docs say to:

> Create a next-drupal project using the basic stater:
>
> ```
> npx create-next-app -e https://github.com/chapter-three/next-drupal-basic-starter
> ```

That means that users will be using whatever commit is at the HEAD of the `main` branch in that repo.

Which is fine. But… how do we do a pre-release version (e.g. 2.0.0-alpha.1) for the starters? As soon as we push code to the `main` branch, any new starters will be using the latest (possibly unstable) code.

### Describe the solution you'd like

For each starter:

1. Leave the default branch as the `main` branch. This will continue to contain the latest _official_ release of the starter.
2. Create a new `canary` branch.
3. Update the `scripts/sync-starters.sh` script to pull and push from the `canary` branch.

Then, to create a new pre-release version:
1. Run the sync script 
2. Go to GitHub and tag a commit with the semantic pre-release label
3. Developers can then use the tagged starter with:
    ```
    npx create-next-app -e https://github.com/chapter-three/next-drupal-basic-starter/tree/2.0.0-alpha.1
    ```

To create a new official release:
1. Run the sync script
2. Go to GitHub and create a PR from `canary` to `main`
3. Merge the PR
4. I don't think we'd need to tag the release since developers can always use the latest official starter with:
    ```
    npx create-next-app -e https://github.com/chapter-three/next-drupal-basic-starter
    ```

### Describe alternatives you've considered

I looked into whether we could use GitHub's "releases" feature, but that just bundles downloadable files. The URLs for those GitHub Releases can't be used by `create-next-app`. `create-next-app` requires a git URL that is clonable, like a branch or tag URL.

**Comment by robdecker** [comment]
### Another option

Instead of releasing alpha versions of the two starters, we can leave the starter repos using the existing 1.6 versions. When people are trying out next-drupal 2.0-alpha, they can use an alpha starter directly from the next-drupal project itself:

```
npx create-next-app -e https://github.com/chapter-three/next-drupal/tree/main/starters/basic-starter
```

This would allow people to use a starter with stable code as default, while enabling users wanting to use/test alpha code to do so as well.

**Comment by JohnAlbin** [comment]
@robdecker Yep. The URL you suggested would work fine with create-next-app.

However, it took me a few days of thinking on this to come up why it made me uneasy. And it has to do with developer wanting to upgrade from the 2.0.0-alpha.1 release to a later release.

GitHub has a `compare/` url that allows you to compare 2 different tags. Inside the monorepo, we can't easily compare 2 tagged starters because GitHub will show all the changes for the entire monorepo between those 2 starter tags. Here's what a 1.7.2 to 1.8.0 comparison shows: https://github.com/chapter-three/next-drupal/compare/basic-starter%401.7.2...basic-starter%401.8.0

If we push to a canary branch and tag releases in the starter-only GitHub repo, we can get clean diffs for the starters and make it easier for developers to upgrade.

**Comment by JohnAlbin** [comment]
Fixed with #641
